StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp

```diff
@@ -0,0 +1,56 @@
+#include "UpgradeRangeLifeSteal.h"
+#include "EngineMath.h"
+#include "ECS/Entity.h"
+#include "Player.h"
+#include "UpgradeMeleeLifeSteal.h"
+#include "Components/MeleeComponent.h"
+
+UpgradeMeleeLifeSteal::UpgradeMeleeLifeSteal(Entity* parent) : Upgrade(parent)
+{
+	// name the upgrade!
+	SetName("UpgradeMeleeLifeSteal");
+	// set the upgrade type/types!
+	SetType(F_UpgradeType::PLAYER);
+	// set the price of this upgrade
+	m_Price = 300;
+	m_StartingPrice = m_Price;
+
+	// percentage of damage done to steal as life
+	m_PercentageGain = 0.10;
+
+	m_ImageName = "MeleeDamage.png";
+}
+
+UpgradeMeleeLifeSteal::~UpgradeMeleeLifeSteal()
+{
+}
+
+void UpgradeMeleeLifeSteal::IncreaseLevel()
+{
+	m_Level++;
+	m_PercentageGain = 0.05f + 0.05f * m_Level; // Starts at 10% and increases by 5% per level
+	m_Price = 300 * pow(m_Level + 1, 2);
```

Comment:
It gets kinda pricy very fast :> tbh I have lost track of how expensive other upgrades get at the moment so maybe it's consistent with other prices.

Reply:
Yeah changed it to m_Price += m_Level * 200
That should make it a bit more afforadable at the first 2-3 levels and then scale pretty well I think.